lib/mix/test/mix/tasks/test_test.exs

```diff
@@ -57,6 +57,32 @@ defmodule Mix.Tasks.TestTest do
     end
   end
 
+  describe "--seed" do
+    test "requires modules alphabetical order if seed is 0" do
+      in_fixture("test_stale", fn ->
+        opts = ["--seed", "0", "--trace"]
+        # Forcing the parallel compiler to operate synchronously
+        envs = [{"ELIXIR_ERL_OPTIONS", "+S 1:1"}]
+        output = mix(["test" | opts], envs)
+
+        matches = Regex.scan(~r"\wTest\s\[.*\]", output)
+        assert matches == [["BTest [test/b_test_stale.exs]"], ["ATest [test/a_test_stale.exs]"]]
+      end)
+    end
+
+    test "requires modules in random order if seed is provided" do
+      in_fixture("test_stale", fn ->
+        opts = ["--seed", "5", "--trace"]
+        # Forcing the parallel compiler to operate synchronously
+        envs = [{"ELIXIR_ERL_OPTIONS", "+S 1:1"}]
+        output = mix(["test" | opts], envs)
+
+        matches = Regex.scan(~r"\wTest\s\[.*\]", output)
+        assert matches == [["ATest [test/a_test_stale.exs]"], ["BTest [test/b_test_stale.exs]"]]
+      end)
+    end
+  end
```

Comment:
Those tests are rather expensive and they rely on implementation details-ish. I am thinking we could just skip them altogether? Other than that, the impl is great!

Reply:
Works for me! Let me remove the test.